### PR TITLE
Issue #735: Event Time support in c++ client

### PR DESCRIPTION
**Pull request by sijie**
This change is part of #732 - adding event time setter and getter in both c++ and python client.

### CODE REVIEW

File: pulsar-client-cpp/include/pulsar/MessageBuilder.h [hunk 1]
```diff
@@ -71,6 +71,11 @@ class MessageBuilder {
     MessageBuilder& setPartitionKey(const std::string& partitionKey);
 
     /**
+     * Set the event timestamp for the message.
+     */
+    MessageBuilder& setEventTimestamp(uint64_t event_timestamp);
```
Comment (merlimat) [inline pulsar-client-cpp/include/pulsar/MessageBuilder.h]: Nitpick: in C++ code we've been using camelCase for classes and var names, like `eventTimestamp`

**Comment by sijie** [comment]
@merlimat cool. I was following the protobuf code convention. I just changed `event_timestamp` to `eventTimestamp`.

File: pulsar-client/src/main/java/org/apache/pulsar/client/api/Message.java [hunk 2]
```diff
@@ -73,9 +73,26 @@
      */
     MessageId getMessageId();
 
+    /**
```
Comment (jai1) [inline pulsar-client/src/main/java/org/apache/pulsar/client/api/Message.java]: Java changes in C++ Client PR?

Comment (jai1) [reply pulsar-client/src/main/java/org/apache/pulsar/client/api/Message.java]: Alternately just merge the JAva PR and the difference won't be seen in this PR.

Comment (sijie) [reply pulsar-client/src/main/java/org/apache/pulsar/client/api/Message.java]: @jai1 this change is based on #734. and they are organized in two different commits. once you merge #734 , this change will be only c++ changes.

Comment (sijie) [reply pulsar-client/src/main/java/org/apache/pulsar/client/api/Message.java]: rebased.

**Comment by merlimat** [comment]
retest this please

**Comment by jai1** [comment]
retest this please

**Comment by merlimat** [comment]
retest this please

**Comment by merlimat** [comment]
retest this please

**Comment by merlimat** [comment]
retest this please

**Pull request merged**